Fix bug secrets escape

jochman opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
handling escape characters from whitelist.json.

## Must have
- [x] Tests

Itay4 commented on demisto_sdk/commands/secrets/tests/secrets_test.py:
```diff
@@ -172,9 +180,17 @@ def test_remove_white_list_regex(self):
+        Random and unmeaningful file content
+        '''
+        file_contents = self.validator.remove_whitelisted_items_from_file(file_contents, white_list)
+        assert file_contents
```
What are we checking here?

jochman replied:
The secret contains `***.url` which is an invalid regex that causes the system to raise an exception.

Itay4 commented:
And how exactly that test checks that?

jochman commented:
> And how exactly that test checks that?

I've found that this term failed when running secrets. So I fixed it and added a unittest to see that it's not failing again.

Itay4 commented:
It is extremely unclear... let’s clarify what we’re testing here in the test docstring and let’s assert something that actually checks for something
